packages/flutter/lib/src/rendering/copy_interceptor.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'selection.dart';
+
+/// Typedef for the inline intercept callback.
+typedef CopyInterceptorCallback = String Function(List<SelectedContent> selections);
+
+/// A handler for intercepting copy operations.
+abstract class CopyInterceptor {
+  /// Creates a [CopyInterceptor].
+  const CopyInterceptor();
```

Comment:
I'm wondering if this needs to be a class instead of just a function, or otherwise if this should be a non-abstract class that takes the `intercept` function as a parameter. Just details to consider.

Reply:
I originally had it as just a function, but it quickly became a mess. Having this as a class has the added benefit of letting us have `const` factory constructors.

Reply:
Also is the `_InlineCopyInterceptor` basically what you're talking about?

Reply:
Yeah thanks, and I guess it's made public with CopyInterceptor.inline 👍